### PR TITLE
Add auto-commit of NOTICE file

### PR DESCRIPTION
**Pull request by artemnikitin**
## Description

This PR adds a Github Action to commit changed NOTICE file

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -46,3 +46,8 @@ jobs:
     - name: Validate the examples
       run: make install validate-examples
       id: validate-examples
+
+    - name: Commit generated files
+      uses: stefanzweifel/git-auto-commit-action@v4
```
Comment (marclop) [inline .github/workflows/go.yml:51]: We're using a different action on other workflows: https://github.com/elastic/terraform-provider-ec/blob/master/.github/workflows/branch.yml#L44-L51. Any reason not to re-use the same one here?

Comment (artemnikitin) [reply .github/workflows/go.yml:51]: Thank you for the suggestion, Marc 👍 

> Any reason not to re-use the same one here?

Not at all, I just missed it. I will change my PR to use this one

Comment (artemnikitin) [reply .github/workflows/go.yml]: Fixed

**Comment by marclop** [comment]
change looks good, could you try to change the NOTICE file (Remove some lines or so) in a commit to see if the GitHub action commits it back?

File: .github/workflows/go.yml (hunk 1 above)
Comment (artemnikitin) [reply .github/workflows/go.yml]: > We're using a different action on other workflows: https://github.com/elastic/terraform-provider-ec/blob/master/.github/workflows/branch.yml#L44-L51. 

I switched to it, but having an issue similar to https://github.com/EndBug/add-and-commit/issues/199. Trying to fix it...

**Comment by marclop** [comment]
It may be easier to just add the NOTICE file generation just like we generate and commit files in the repository branches (`master` and version branches `x.x`). Seems like that GitHub action is not very happy to commit on forked repositories.

File: .github/workflows/changelog_generation.yml [hunk 2]
```diff
@@ -20,13 +20,15 @@ jobs:
...
+      - run: make changelog vendor
       - name: Commit & Push changes
         run: |
           if [[ ! -z  $(git status --porcelain) ]]; then
             git config --local user.email elasticcloudclients@elastic.co
             git config --local user.name elasticcloudclients
             git add CHANGELOG.md
             git commit -m "Update CHANGELOG.md"
+            git add NOTICE
+            git commit -m "Update NOTICE"
```
Comment (marclop) [inline .github/workflows/changelog_generation.yml:32]: I'm not sure this approach will work very well when there aren't any changelog updates but we need to re-generate the NOTICE file. It's probably not a big deal, but this means that the NOTICE file will only be ever updated when the changelog has new entries.

```console
$ git commit -m "test"; echo $?
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
1
```

I would instead do add both files on the `git add` on line 29 and commit both files with an updated commit message.

Comment (artemnikitin) [reply .github/workflows/changelog_generation.yml]: Fixed

**Review by marclop — APPROVED** [review]
Awesome, Thank you for all the work Artem. I know testing this wasn't the easiest!

**Pull request merged**